### DISCUSSION TITLE
running mojo files

### DISCUSSION THREAD

**Discussion by BlackHo0d** [modular/modular#1118] [General]
I've installed mojo and now can use it in the REPL. But I can't run mojo files. I've read that I should add the following lines to your configuration file (.bash_profile, .bashrc, or .zshrc) but I don't understand how to do it. Can anyone help me with that ?

I'm using Mac OS(Intel)

**Comment by rarebreed** [comment]
There are two ways you can run a mojo file.  One is through the `mojo` "interpreter", and the other is to build an executable and run it natively.

```python
# cli.mojo
import sys

fn main():
    let args = sys.argv()
    for idx in range(1, len(args)):
        print(args[idx])
```
Now you can run it through the mojo command without building an executable

```bash
❯ : mojo cli.mojo hello world
hello
world

# You can also do `mojo run`
❯ : mojo run cli.mojo hello world
hello
world
```

Or you can build a native executable

```bash
~/Projects/anting 
❯ : mojo build cli.mojo

~/Projects/anting 
❯ : ./cli hello world
hello
world
```

**Reply by rarebreed** [reply]
By the way, there's some documentation if you haven't read it yet.

https://docs.modular.com/mojo/manual/get-started/hello-world.html#build-and-run-mojo-source-files

- How to [build an executable](https://docs.modular.com/mojo/cli/build.html)
- How to [run a mojo file](https://docs.modular.com/mojo/cli/run.html)
